addons/sale_project/models/project.py

```diff
@@ -177,6 +179,18 @@ def _compute_sale_line(self):
             if task.sale_line_id.order_partner_id.commercial_partner_id != task.partner_id.commercial_partner_id:
                 task.sale_line_id = False
 
+    @api.depends('sale_order_id')
+    def _compute_display_sale_order_button(self):
+        if not self.sale_order_id:
+            self.display_sale_order_button = False
+            return
+        try:
+            sale_orders = self.env['sale.order'].search([('id', 'in', self.sale_order_id.ids)])
+            for task in self:
+                task.display_sale_order_button = task.sale_order_id and task.sale_order_id in sale_orders
```

Comment:
Do we really need to check that `task.sale_order_id` is not Falsy prior to check if in `sale_orders`?